### DISCUSSION TITLE
I want to know what is the difference between this library and pyroms. The author of pyroms is no longer updating

### DISCUSSION THREAD

**Discussion by q1165600785** [CWorthy-ocean/roms-tools#96] [General]
pyroms is also a library that creates ROMS output. I'd like to know if there is a difference between these two

**Comment by NoraLoose** [comment]
Hi @q1165600785, thanks for your question! We are building `ROMS-Tools` because we think that we need a modern python package that creates input files for ROMS. 

While I have never used `pyroms` (so I am by no means an expert on that library), I would think there are a few key differences:
* **Documentation:** `ROMS-Tools` puts a lot of emphasis on an [extensive documentation](https://roms-tools.readthedocs.io/en/latest/) to ensure high user-friendliness
* **Installation:** To ensure user-friendliness, we want `ROMS-Tools` to be easily installable. Currently, you can install it from pip (see our documentation page), and soon it will be available on conda as well.
* **Maintenance**: As you say, it seems that `pyroms` is no longer maintained. We are still building (and will maintain) `ROMS-Tools`.
* Note that `ROMS-Tools` is still under development, and not fully operational and tested yet - but should be in a few weeks to months, stay tuned!
* We are currently focusing on ROMS input files for [UCLA-ROMS](https://github.com/CESR-lab/ucla-roms/issues) but may make an effort to make `ROMS-Tools` compatible with the other ROMS versions in the future.

Let us know what you think!

**Comment by q1165600785** [comment]
Thank you for the author's response. I would also like to know what are the differences between UCLA Roms and Rutgers University Roms. I would like to use UCLA Roms to simulate tidal processes in the Sea of Japan region. When I use UCLA Roms for modeling, do I only need tidal forcing without surface and boundary forcing and initial conditions, or are all three forcing and one initial condition necessary

**Reply by NoraLoose** [reply]
This sounds like a great question for the UCLA ROMS and Rutgers ROMS developers! You probably know the links to their repos, but I am posting them here just for reference:
* UCLA-ROMS: https://github.com/CESR-lab/ucla-roms
* Rutgers ROMS: https://github.com/myroms/roms (I think)

**Reply by q1165600785** [reply]
I feel the author's prompt response. May I ask if roms tools can now generate roms input files for simulation or if more extensive testing is needed before input file production can be carried out for simulation

**Reply by NoraLoose** [reply]
As I said above

> Note that ROMS-Tools is still under development, and not fully operational and tested yet - but should be in a few weeks to months, stay tuned!

ROMS-Tools is still under development and in the testing phase.

**Reply by q1165600785** [reply]
Professor, I have reviewed the input files required by UCLA Roms. May I ask if Roms tools can add a feature to generate tides. in tides. opt files, which are input files for production parameters? I'm not sure if Professor can understand what I mean

**Reply by NoraLoose** [reply]
ROMS-Tools already supports the generation of tidal forcing, see https://roms-tools.readthedocs.io/en/latest/tides.html.

**Reply by q1165600785** [reply]
Professor, can you provide an example to illustrate the simulation of tidal processes in a region? As a beginner, I don't seem to understand some of the things involved

**Reply by q1165600785** [reply]
Professor, if I were to simulate the Bohai Sea region in China, and I only want the eastern boundary to be the open boundary and the other boundaries to be the closed boundary, and use the heights of the 8 tides of tpxo as forcing for the eastern boundary, what should I do
